src/book/02-system/01-nubots/02-build-system.mdx

```diff
@@ -272,6 +272,62 @@ Be sure that you are using the `generic` target for this (`./b target generic`),
 
 </Alert>
 
+The `run` command provides some options to help you debug your roles.
+
+<Alert type="warning">
+
+Be sure that you are using the `generic` target for this (`./b target generic`). You will likely want to recompile your role after setting `CMAKE_BUILD_TYPE` to either `Debug` or `RelWithDebInfo` otherwise you will likely get useless information.
+
+</Alert>
+
+
+| Subcommand     | Description                                                                              |
+| -------------- | ---------------------------------------------------------------------------------------- |
+| --asan         | Enables address sanitisation, a fast memory error detector.                              |
+| --gdb          | Runs selected role inside of GDB to enable debugging.                                    |
+| --valgrind     | Runs selected role inside of valgrind to enable detection of memory errors.              |
+
+In order to use ASan you must recompile your role after setting `USE_ASAN` to `ON` in the cmake options (`./b configure -i`).
+
+Currently, using the `--asan` option adds an environment variable that will direct ASan to log all of its output to a file. This is done to ensure that even when you are running a curses-enabled role (e.g. `keyboardwalk`) you will always be able to revocer the output from ASan (and it will also be clean from other log messages). There are a number of other run-time options you could set to tune the behaviour of ASan, they can be found [here](https://github.com/google/sanitizers/wiki/AddressSanitizerFlags). Just run your role as follows
```

Comment:
```suggestion
Currently, using the `--asan` option adds an environment variable that will direct ASan to log all of its output to a file. This is done to ensure that even when you are running a curses-enabled role (e.g. `keyboardwalk`) you will always be able to recover the output from ASan (and it will also be clean from other log messages). There are a number of other run-time options you could set to tune the behaviour of ASan, they can be found [here](https://github.com/google/sanitizers/wiki/AddressSanitizerFlags). Just run your role as follows:
```